src/features/all-drops/components/AllDrops.tsx

```diff
@@ -193,16 +193,38 @@ export default function AllDrops() {
     [pagination],
   );
 
-  useEffect(() => {
+  const handleMultipleEffects = async () => {
     if (!accountId) return;
+
     // page query param should be indexed from 1
     const pageQuery = searchParams.get('page');
     const currentPageIndex = pageQuery !== null ? parseInt(pageQuery) - 1 : 0;
     setPagination((pagination) => ({ ...pagination, pageIndex: currentPageIndex }));
     handleGetDropsSize();
     handleGetDrops({ start: currentPageIndex * pagination.pageSize });
-    handleFinishNFTDrop();
-  }, [accountId, searchParams]);
+
+    // first mount, check for NFT attempt
+    if (refreshTrigger === '') {
+      setAppModal({
+        isOpen: true,
+        isLoading: true,
+        header: 'Creating NFT',
+        message:
+          'Uploading media and creating NFT drop links on-chain. This may take 15-30 seconds.',
+      });
+      await handleFinishNFTDrop(setRefreshTrigger, setAppModal);
```

Comment:
In `CreateNftDropContext`, Could we move `handleFinishNFTDrop` and `createDropsForNFT` to a hook?

Advantages are:
1. You don't have to pass setAppModal and setRefreshTrigger, they can be obtained using the context hook 
```
const { setAppModal } = useAppContext()
```
2. All drops page are including unused libraries declared in `CreateNftDropContext`, so all drops page are adding unnecessary network load
![image](https://user-images.githubusercontent.com/19740800/228433193-153e2d8b-fb0c-4dc4-a93a-fb10ddb1f315.png)
In All drops page (after production build),
![image](https://user-images.githubusercontent.com/19740800/228433345-a1f61f31-cd9f-4347-9fee-408153381cf1.png)

Reply:
Yes will do this to support ticket gift NFT as well